nixos/doc/manual/release-notes/rl-2311.section.md

```diff
@@ -606,3 +601,95 @@ The module update takes care of the new config syntax and the data itself (user
+  A partial and basic implementation of GVariant formatted strings.
+  See [GVariant Format Strings](https://docs.gtk.org/glib/gvariant-format-strings.html) for details.
+
+  :::{.warning}
+  This API is not considered fully stable and it might therefore
+  change in backwards incompatible ways without prior notice.
+  :::
+
+### New functions {#sec-release-23.11-lib-new-functions}
+
+- `lib.asserts`:
+  [`assertEachOneOf`](https://nixos.org/manual/nixpkgs/unstable#function-library-lib.asserts.assertEachOneOf)
+- `lib.attrsets`:
+  [`attrsToList`](https://nixos.org/manual/nixpkgs/unstable#function-library-lib.attrsets.attrsToList)
+- `lib.customisation`:
+  [`makeScopeWithSplicing'`](https://nixos.org/manual/nixpkgs/unstable#function-library-lib.customisation.makeScopeWithSplicing-prime)
+- `lib.generators`:
+  [`mkDconfKeyValue`](https://nixos.org/manual/nixpkgs/unstable#function-library-lib.generators.mkDconfKeyValue),
+  [`toDconfINI`](https://nixos.org/manual/nixpkgs/unstable#function-library-lib.generators.toDconfINI)
+- `lib.lists`:
+  [`findFirstIndex`](https://nixos.org/manual/nixpkgs/unstable#function-library-lib.lists.findFirstIndex),
+  [`hasPrefix`](https://nixos.org/manual/nixpkgs/unstable#function-library-lib.lists.hasPrefix),
+  [`removePrefix`](https://nixos.org/manual/nixpkgs/unstable#function-library-lib.lists.removePrefix),
+  [`commonPrefix`](https://nixos.org/manual/nixpkgs/unstable#function-library-lib.lists.commonPrefix),
+  [`allUnique`](https://nixos.org/manual/nixpkgs/unstable#function-library-lib.lists.allUnique)
+- `lib.path`:
+  [`hasPrefix`](https://nixos.org/manual/nixpkgs/unstable#function-library-lib.path.hasPrefix),
+  [`removePrefix`](https://nixos.org/manual/nixpkgs/unstable#function-library-lib.path.removePrefix),
+  [`splitRoot`](https://nixos.org/manual/nixpkgs/unstable#function-library-lib.path.splitRoot),
+  [`subpath.components`](https://nixos.org/manual/nixpkgs/unstable#function-library-lib.path.subpath.components)
+- `lib.strings`:
+  [`replicate`](https://nixos.org/manual/nixpkgs/unstable#function-library-lib.strings.replicate),
+  [`cmakeOptionType`](https://nixos.org/manual/nixpkgs/unstable#function-library-lib.strings.cmakeOptionType),
+  [`cmakeBool`](https://nixos.org/manual/nixpkgs/unstable#function-library-lib.strings.cmakeBool),
+  [`cmakeFeature`](https://nixos.org/manual/nixpkgs/unstable#function-library-lib.strings.cmakeFeature)
+- `lib.trivial`:
+  [`mirrorFunctionArgs`](https://nixos.org/manual/nixpkgs/unstable#function-library-lib.trivial.mirrorFunctionArgs)
+- `lib.systems`:
+  [`equals`](https://nixos.org/manual/nixpkgs/unstable#function-library-lib.systems.equals)
+
+### Function changes {#sec-release-23.11-lib-function-changes}
+
+- `lib.meta.getExe pkg` now gives a warning if `pkg.meta.mainProgram` is not set,
+  but it continues to default to the derivation name.
+  Use `lib.meta.getExe' pkg "someBinary"` to avoid the warning and/or select a different binary.
```

Comment:
```suggestion
  Nixpkgs accepts PRs that set `meta.mainProgram` on packages where it makes sense.
  Use `lib.meta.getExe' pkg "some-command"` to avoid the warning and/or select a different executable.
```

Ideally we have better data: more mainPrograms.
Nits:
- commands tend to be lower snake case
- avoid "binary" because scripts aren't binaries (`/bin` is very very unfortunate)

Reply:
`lib.meta.getExe` is annoying. Do we really want people to type that?
Sublibraries are cool, but everyone writes `lib.getExe`.